### PR TITLE
Migrate sample sheet models

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR migrates the sample sheet models to Pydantic V2.

- Replace deprecated `Config` with [`model_config`](https://docs.pydantic.dev/2.0/migration/#changes-to-config) attribute
- Replace deprecated `parse_obj_as` with [`TypeAdapter`](https://docs.pydantic.dev/2.0/migration/#introduction-of-typeadapter)
- Replace import of `BaseModel` and `ValidationError`

### Fixed
- Migrate sample sheet Pydantic models to V2

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/fixtures/apps/demultiplexing/flow_cells_demux_all/211101_A00187_0615_AHLG5GDRXY/SampleSheet.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [Data]
-FCID,Lane,SampleID,SampleRef,index,index2,SampleName,Control,Recipe,Operator,Project
```
Comment (seallard) [inline tests/fixtures/apps/demultiplexing/flow_cells_demux_all/211101_A00187_0615_AHLG5GDRXY/SampleSheet.csv:2]: The flow cell seems to have been sequenced on a novaseqx machine but the structure of the sample sheet seems to be for bcl2fastq (?) Changed it to match the structure for bclconvert.

Comment (seallard) [reply tests/fixtures/apps/demultiplexing/flow_cells_demux_all/211101_A00187_0615_AHLG5GDRXY/SampleSheet.csv:2]: This caused a test to fail when the pydantic model was migrated, not sure why it does not fail in master. I don't think it is a case we support currently anyway (running bcl2fastq on novaseqx)

**Review by diitaz93 — APPROVED** [review]
Great 💯

File: cg/apps/demultiplex/sample_sheet/models.py [hunk 2]
```diff
@@ -15,10 +15,7 @@ class FlowCellSample(BaseModel):
     sample_id: str
     index: str
     index2: str = ""
-
-    class Config:
-        allow_population_by_field_name = True
-        extra = Extra.ignore
+    model_config = ConfigDict(populate_by_name=True, extra="ignore")
```
Comment (islean) [inline cg/apps/demultiplex/sample_sheet/models.py:18]: Why did we switch from Extra.ignore?

**Review by islean — APPROVED** [review]
Nice 🔥

**Pull request merged**